src/lib/users/leaf_blk_device.rb

```diff
@@ -0,0 +1,121 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "json"
+require "yast2/execute"
+
+module Y2Users
+  # This class represents a block device reported by `lsblk` as a leaf node
+  #
+  # As we cannot use a devicegraph for the time being, this class extracts the
+  # block devices information from the lsblk command. Only the leaf devices
+  # are taken into account and it only includes information which is relevant
+  # for the {Y2Users::Widgets::DiskSelector} widget.
+  class LeafBlkDevice
+    class << self
+      # Returns all relevant block devices
+      #
+      # @note It takes the information from `lsblk`
+      #
+      # @return [Array<LeafBlkDevice>] List of relevant block devices
+      def all
+        lsblk["blockdevices"].map { |h| new_from_hash(h) }
+      end
+
+      # Instantiates a new object
+      #
+      # @note It uses a Hash with information from `lsblk`.
+      #
+      # @return [LeafBlkDevice] New LeafBlkDevice instance
+      def new_from_hash(hash)
+        parent = find_root_device(hash)
+        new(
+          name: hash["name"], disk: parent["name"], model: parent["model"],
+          transport: parent["tran"], fstype: hash["fstype"]
+        )
+      end
+
+    private
+
+      # Gets `lsblk` into a Hash
+      #
+      # @return [Hash] Hash containing data from `lsblk`
+      def lsblk
+        output = Yast::Execute.locally(
+          "/usr/bin/lsblk", "--inverse", "--json", "--paths",
+          "--output", "NAME,TRAN,FSTYPE,MODEL", stdout: :capture
+        )
+        JSON.parse(output)
```

Comment:
`Yast::Execute.locally` returns `nil` in case of failure. Then `JSON.parse(nil)` will crash here... :boom: